packages/renderer/src/lib/ui/TreeView.svelte

```diff
@@ -0,0 +1,130 @@
+<style>
+.no-arrow {
+  padding-left: 1rem;
+}
+.arrow {
+  cursor: pointer;
+  display: inline-block;
+}
+.arrowDown {
+  transform: rotate(90deg);
+}
+</style>
+
+<script context="module" lang="ts">
+const _expansionState = new Map<string, boolean>();
+</script>
+
+<script lang="ts">
+import type { fileNode } from '../../../../main/src/plugin/filetree';
+
+export let tree: fileNode<any>;
+export let margin = 0;
+export let root = true;
+
+$: label = isRemoved(tree) ? tree.name.substring(4) : tree.name;
+$: children = tree.children;
+$: file = tree.data;
+$: colorClass = getColor(tree.name, tree.data);
+
+function getColor(name: string, data: any) {
+  if (!data) {
+    return '';
+  }
+  if (name.startsWith('.wh.')) {
+    return 'text-red-500';
+  }
+  if (data.type === 'SymbolicLink') {
+    return 'text-sky-300';
+  }
+  if (data.type === 'Directory') {
+    return 'text-sky-500';
+  }
+  if (isExecutable(data.mode)) {
+    return 'text-green-500';
+  }
+  return '';
+}
+
+function isRemoved(data: any): boolean {
+  return data?.name.startsWith('.wh.');
+}
+
+$: expanded = _expansionState.get(label) || false;
+const toggleExpansion = () => {
+  expanded = !expanded;
+  _expansionState.set(label, expanded);
+};
+$: arrowDown = expanded;
+function getModeString(type: string, mode: number): string {
+  return (
+    (type === 'Directory' ? 'd' : '-') +
+    (mode & 0o400 ? 'r' : '-') +
+    (mode & 0o200 ? 'w' : '-') +
+    (mode & 0o100 ? 'x' : '-') +
+    (mode & 0o040 ? 'r' : '-') +
+    (mode & 0o020 ? 'w' : '-') +
+    (mode & 0o010 ? 'x' : '-') +
+    (mode & 0o004 ? 'r' : '-') +
+    (mode & 0o002 ? 'w' : '-') +
+    (mode & 0o001 ? 'x' : '-')
+  );
+}
+
+function isExecutable(mode: number): boolean {
+  return (mode & 0o111) !== 0;
+}
+
+function getHumanSize(size: number): string {
+  let u = '';
+  if (size > 1024) {
+    size = Math.floor(size / 100) / 10;
+    u = 'k';
+  }
+  if (size > 1024) {
+    size = Math.floor(size / 100) / 10;
+    u = 'M';
+  }
+
+  if (size > 99) {
+    size = Math.floor(size);
+  }
+  return size + u;
+}
```

Comment:
it can't go into a generic TreeView component, it looks like a specific TreeImageLayer